ui/src/components/search/search-input.ts

```diff
@@ -0,0 +1,706 @@
+
+import '@geoblocks/ga-search';
+import {getLayersConfig, SwisstopoImageryLayersConfig} from '../../swisstopoImagery';
+import {Cartographic, Entity as CesiumEntity, Math as CesiumMath, Rectangle, Viewer as CesiumViewer} from 'cesium';
+import {lv95ToDegrees} from '../../projection';
+import {escapeRegExp} from '../../utils';
+import {extractEntitiesAttributes} from '../../query/objectInformation';
+import auth from '../../store/auth';
+import defaultLayerTree, {LayerTreeNode} from '../../layertree';
+import NavToolsStore from '../../store/navTools';
+import {SearchLayer, SearchLayerWithLayer, SideBar} from '../../elements/ngm-side-bar'; // <ga-search> component
+import {createRef, ref} from 'lit/directives/ref.js';
+
+@customElement('ngm-search-input')
+export class SearchInput extends LitElementI18n {
+  @property()
+  private accessor viewer: CesiumViewer | null = null
+
+  @property()
+  private accessor sidebar: SideBar | null = null
+
+  @state()
+  private accessor isActive = false
+
+  private accessor searchRef = createRef<HTMLElement>()
+  private accessor inputRef = createRef<HTMLInputElement>()
+  private accessor resultsRef = createRef<HTMLUListElement>()
+
+  private layerConfigs: SwisstopoImageryLayersConfig | null = null;
+
+  constructor() {
+    super();
+    this.initialize();
+
+    // Bind `this` for callback methods.
+    this.clear = this.clear.bind(this);
+    this.handleItemSelected = this.handleItemSelected.bind(this);
+    this.handleInputKeyPress = this.handleInputKeyPress.bind(this);
+    this.matchFeature = this.matchFeature.bind(this);
+    this.renderItem = this.renderItem.bind(this);
+    this.searchAdditionalItems = this.searchAdditionalItems.bind(this);
+    this.toggleActive = this.toggleActive.bind(this);
+  }
+
+  private initialize(): void {
+    getLayersConfig().then((layersConfig) => {
+      this.layerConfigs = layersConfig;
+    });
+  }
+
+  render = () => html`
+    <ga-search
+      ${ref(this.searchRef)}
+      types="location,additionalSource,layer,feature"
+      locationOrigins="zipcode,gg25,gazetteer"
+      .filterResults="${this.matchFeature}"
+      .renderResult="${this.renderItem}"
+      .additionalSource="${{
+        search: this.searchAdditionalItems,
+        getResultValue: (item: AdditionalItem) => item.label,
+      }}"
+      @submit="${this.handleItemSelected}"
+    >
+      <!-- Freeze the event's target to this input, as otherwise, ga-input is unable to read the input's value. -->
+      <input
+        ${ref(this.inputRef)}
+        type="search"
+        placeholder="${i18next.t('header_search_placeholder')}"
+        @input="${freezeEventTarget}"
+        @click="${captureEvent}"
+        @keydown="${this.handleInputKeyPress}"
+      >
+      <ul ${ref(this.resultsRef)}></ul>
+    </ga-search>
+
+    <span class="icon is-close" @click="${this.clear}"></span>
+    <span class="icon is-search" @click="${this.toggleActive}"></span>
```

Comment:
This is for mobile only, where the search is only visible after clicking the search icon. On desktop, it should just focus the search input. Does something else happen for you (e.g. flickering)? I was not able to reproduce it as of yet.

Reply:
Maybe flickering is the wrong word, but when it is in focus already and i click it again, the results quickly disappear (while hodling the mouse) and then reappear. BUt this might just be browser behavior